pydantic/fields.py

```diff
@@ -356,14 +357,27 @@ class MyModel(pydantic.BaseModel):
     def merge_field_infos(*field_infos: FieldInfo, **overrides: Any) -> FieldInfo:
         """Merge `FieldInfo` instances keeping only explicitly set attributes.
 
+        Later `FieldInfo` instances override earlier ones.
+
         Returns:
             FieldInfo: A merged FieldInfo instance.
         """
+        flattened_field_infos: list[FieldInfo] = []
+        for field_info in field_infos:
+            flattened_field_infos.extend(x for x in field_info.metadata if isinstance(x, FieldInfo))
+            flattened_field_infos.append(field_info)
+        field_infos = tuple(flattened_field_infos)
         new_kwargs: dict[str, Any] = {}
+        metadata = {}
         for field_info in field_infos:
             new_kwargs.update(field_info._attributes_set)
+            for x in field_info.metadata:
+                if not isinstance(x, FieldInfo):
+                    metadata[type(x)] = x
```

Comment:
I ran into a case where this was necessary to get proper merging of FieldInfos that use `gt`, `lt`, etc.

Reply:
This stuff is getting really annoying. @samuelcolvin not having to do this stuff is one of the big advantages of the Output enum thing we were discussing the other day.